pkg/sigv4/sigv4.go

```diff
@@ -210,6 +210,10 @@ func createSigner(cfg *Config, verboseMode bool) (*v4.Signer, error) {
 			s.Debug = aws.LogDebugWithSigning
 		}
 	}
+	assumeRoleRegion := aws.String(cfg.Region)
+	if awsds.IsOptInRegion(cfg.Region) {
+		assumeRoleRegion = aws.String("us-east-1")
```

Comment:
```suggestion
        // AMG: Currently the auth middleware for Prometheus doesn’t support assuming role to opt-in regions, so we query from non-opt-in region.
        // This will be superseded by proper opt-in regions support for prometheus in a later version.
        // https://github.com/grafana/grafana/issues/107199
		assumeRoleRegion = aws.String("us-east-1")
```